drizzlepac/haputils/poller_utils.py

```diff
@@ -1096,33 +1305,43 @@ def build_poller_table(input: str, log_level, all_mvm_exposures=[], poller_type=
     # viable, it should not be included in the output "poller" table.
     # NOTE: The "all_mvm_exposures" variable does not need to be checked for "usable" datasets
     # as it is used to create a WCS/footprint to accommodate *all* exposures in an MVM visit.
-    usable_datasets, return_code = analyze.analyze_wrapper(datasets, use_sbchrc=include_small, type=poller_type)
+    usable_datasets, return_code = analyze.analyze_wrapper(
+        datasets, use_sbchrc=include_small, type=poller_type
+    )
     if not usable_datasets:
-        log.warning("No usable images in poller file or input list for drizzling. The processing of this data is ending.")
+        log.warning(
+            "No usable images in poller file or input list for drizzling. The processing of this data is ending."
+        )
         sys.exit(return_code)
     else:
-        log.info("There are {} usable images identified in the poller file for processing.".format(len(usable_datasets)))
+        log.info(
+            "There are {} usable images identified in the poller file for processing.".format(
+                len(usable_datasets)
+            )
+        )
 
     cols = OrderedDict()
     for cname in poller_colnames:
         cols[cname] = []
-    cols['filename'] = usable_datasets
+    cols["filename"] = usable_datasets
     if input_table:
-        if 'aperture' in input_table.colnames:
-            cols['aperture'] = input_table['aperture'].tolist()
+        if "aperture" in input_table.colnames:
+            cols["aperture"] = input_table["aperture"].tolist()
         else:
-            add_col = Column(['empty_aperture'] * len(usable_datasets), name='aperture', dtype='str')
+            add_col = Column(
+                ["empty_aperture"] * len(usable_datasets), name="aperture", dtype="str"
+            )
             input_table.add_column(add_col, index=7)
```

Comment:
There is a failure when trying to add a column when the number of usable datasets and the length of the input table are not the same.  This can be true for **_many_** reasons.   The job of the analyze module is to identify and return only the datasets which can actually be used during processing.

Reply:
add_col = Column(
                ["empty_aperture"] * len(input_table), name="aperture", dtype="str"
            )

Reply:
Issue should be resolved in [most recent commit](https://github.com/spacetelescope/drizzlepac/pull/1693/commits/511fa54a9808d8b5e8c8ef5a6ce0dfe395c5d8e9)